### ISSUE TITLE
MediaConvert - Disable encryption in output group 

### ISSUE THREAD

**Issue by MkJha** [aws/aws-sdk-java-v2#1598]
I don't find any flag in the OutputGroup groupsettings object to disable the DRM encryption.

For example : I want to disable HlsGroupSettings drm encryption HlsEncryptionSettings.

One way could be to not to create this encrytion object while creating the output settings. 

But in our case, we want to use the template outputgroup and override a few details. We are overriding using the copy method copy(Consumer<? super B> modifier). However it does not allow to set any child attribute as null while copying.

It boils down to two question :: 

a. Firstly there should be just a flag indicating whether DRM settings is enabled or not (just like AWS console)
b. Or there must be some way to remove a object/set it to null while copying , which it does not allow: 
example :: hlsGroupSettings = templateHlsGroupSettings.copy(x->x.destination(destinationUrl)).copy(x->x.encryption(hlsEncryptionSettings)) // I should be able to remove the existing drm encrytpion setting/or set it to null

**Comment by debora-ito** [comment]
>But in our case, we want to use the template outputgroup and override a few details. We are overriding using the copy method copy(Consumer<? super B> modifier). However it does not allow to set any child attribute as null while copying.

 @MkJha can you provide a code sample showing what you tried to do?

**Comment by debora-ito** [comment]
@MkJha I reached out to the MediaConvert team and they said it's not possible to remove a setting element that is referenced in a job template. You can only override or add parameters to settings that exists.

A suggestion is to work with a base template that does not reference DRM, and then use parameter overrides to set it when needed in the job creation.

Gonna go ahead and close this since there's no issue with the SDK. Please feel free to reopen if you have any other question.

**Comment by MkJha** [comment]
Let me rephrase my question , You have a template already created which has drm encryption enabled and you want to update the template with drm encryption disabled? SO basically its boils down to updating an entity. So the basic update mechanism would be do a get and update the attributes you want. Isn't it expected here as well?

**Comment by debora-ito** [comment]
After a quick look at the API, you should be able to update the template creating an [UpdateJobTemplateRequest](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/mediaconvert/model/UpdateJobTemplateRequest.html) and redefining the OutputGroupSettings 
 in the [JobTemplateSettings](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/mediaconvert/model/JobTemplateSettings.html) to not have a DRM encryption on its [HlsGroupSettings](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/mediaconvert/model/HlsGroupSettings.Builder.html).

If this is not what you are experiencing, please provide us a sample code of what you are trying to do and the results you are getting.

In addition, for general AWS Services guidance we recommend asking on the [AWS Developer Forums](https://forums.aws.amazon.com/forum.jspa?forumID=266) or StackOverflow to get support from the community.

**Comment by MkJha** [comment]
As i have mentioned earlier scenario is fetching the template(Suppose with one output group with DRM enabled).
I just want to get the output group from the retrieved template and disable drm and use it my job settings. Hence tmplate remains intact.
